girder/constants.py

```diff
@@ -200,7 +200,76 @@ class TokenScope:
     ASSETSTORES_READ = 'core.assetstore.read'
     PARTIAL_UPLOAD_READ = 'core.partial_upload.read'
     PARTIAL_UPLOAD_CLEAN = 'core.partial_upload.clean'
-    DATA_READ = 'core.data_read'
+    DATA_READ = 'core.data.read'
+    DATA_WRITE = 'core.data.write'
+    DATA_ADMIN = 'core.data.admin'
+    USER_INFO_READ = 'core.user_info.read'
+
+    _customScopes = []
+    _adminCustomScopes = []
+
+    @classmethod
+    def describeScope(cls, scopeId, name, description, admin=False):
+        """
+        Register a description of a scope.
+
+        :param scopeId: The unique identifier string for the scope.
+        :type scopeId: str
+        :param name: A short human readable name for the scope.
+        :type name: str
+        :param description: A more complete description of the scope.
+        :type description: str
+        :param admin: If this scope only applies to admin users, set to True.
+        :type admin: bool
+        """
+        info = {
+            'id': scopeId,
+            'name': name,
+            'description': description
+        }
+        if admin:
+            cls._adminCustomScopes.append(info)
+        else:
+            cls._customScopes.append(info)
+
+    @classmethod
+    def listScopes(cls):
+        return {
+            'custom': cls._customScopes,
+            'adminCustom': cls._adminCustomScopes
+        }
+
+TokenScope.describeScope(
+    TokenScope.USER_INFO_READ, 'Read your user information',
+    'Allows clients to look up your user information, including private fields '
+    'such as email address.')
+TokenScope.describeScope(
+    TokenScope.DATA_READ, 'Read data',
+    'Allows clients to read all data that you have access to.')
+TokenScope.describeScope(
+    TokenScope.DATA_WRITE, 'Write data',
+    'Allows clients to edit data in the hierarchy and create new data anywhere '
+    'you have write access.')
+TokenScope.describeScope(
+    TokenScope.DATA_ADMIN, 'Data administration', 'Allows administrative '
+    'control on data you own, including setting access control and deletion.'
+)
+
+TokenScope.describeScope(
+    TokenScope.PLUGINS_ENABLED_READ, 'See enabled plugins', 'Allows clients '
+    'to see the list of plugins enabled on the server.', admin=True)
+TokenScope.describeScope(
+    TokenScope.SETTINGS_READ, 'See system setting values', 'Allows clients to '
+    'view the value of any system setting.', admin=True)
+TokenScope.describeScope(
+    TokenScope.ASSETSTORES_READ, 'View assetstores', 'Allows clients to see '
+    'all assetstore information.', admin=True)
+TokenScope.describeScope(
+    TokenScope.PARTIAL_UPLOAD_READ, 'View unfinished uploads.',
+    'Allows clients to see all partial uploads.', admin=True)
+TokenScope.describeScope(
+    TokenScope.PARTIAL_UPLOAD_CLEAN, 'Remove unfinished uploads.',
+    'Allows clients to remove unfinished uploads.', admin=True)
```

Comment:
It seems that `admin=True` here can be confused with the naming `TokenScope.DATA_ADMIN`, and in general the admin term should be reserved for administrator-account things. I wonder if you should call the token something like `TokenScope.DATA_FULL_ACCESS` so that developers and users aren't confused seeing the word "administration" on something that does not require an administrative account.